src/Avalonia.Controls/VirtualizingStackPanel.cs

```diff
@@ -787,7 +787,7 @@ public IReadOnlyList<double> GetIrregularSnapPoints(Orientation orientation, Sna
                     {
                         var averageElementSize = EstimateElementSizeU();
                         double snapPoint = 0;
-                        for (var i = 0; i < Items.Count; i++)
+                        for (var i = 0; i < itemCount; i++)
```

Comment:
We only need to check realized items, right? In that case would it make more sense to only iterate over the items in `_realizedElements` instead of iterating through all items in the source?